### PR TITLE
fix(input): emit input event

### PR DESCRIPTION
**Pull request by devopsmariocom**

### CODE REVIEW

File: packages/input/Input.tsx [hunk 1]
```diff
@@ -57,14 +57,16 @@ export class Input extends Component<InputProps> {
 
   inputElement: HTMLInputElement;
 
-  private provideValue(event: Event) {
+  private delegateEventToParent( event: Event ) {
+
     this.value = this.inputElement.value;
-    emit(this,'change'); // emit change event on root element
+    emit(this, event.type);
```
Comment (Hotell) [inline packages/input/Input.tsx:63]: this is wrong implementation. oninput events are propagated from shadow boundaries. by doing this we are triggering double oninput event

File: packages/input/index.test.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+</head>
+<body>
+
+  <bl-input id="getValue">
+
+  <script>
+
+    suite('<bl-input>', function() {
+
+      var getValue = document.getElementById( 'getValue' );
```
Comment (Hotell) [inline packages/input/index.test.html:19]: this is very poor and confusing naming

File: packages/input/index.test.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        expect(getValue.value).to.be.equal('c');
+
+      });
+
+      test('emit input', function( done ) {
+
+        var getValueInput = getValue.shadowRoot.querySelector('input');
+
+        getValue.addEventListener( 'input', function() {
```
Comment (Hotell) [inline packages/input/index.test.html:41]: this test is wrong. you wanna listen on custom element, not on it shadow dom private workings

File: packages/input/index.test.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        getValueInput.dispatchEvent( new Event( 'input' ) );
+
+      });
+
+      test('emit change', function( done ) {
+
+        var getValueInput = getValue.shadowRoot.querySelector('input');
+
+        getValue.addEventListener( 'change', function() {
```
Comment (Hotell) [inline packages/input/index.test.html:55]: same issue as previous comment

**Review by Hotell — CHANGES_REQUESTED** [review]
My concerns were confirmed by doing this CR. I have to  be part of this development ASAP 🔢

**Comment by devopsmariocom** [comment]
dear @Hotell tests are in progress(bore), if you have time please feel free to contribute to this project, because it's really easy to judge work of others from perspective of god 😉.

**Pull request closed without merging**